src/sentry/static/sentry/app/components/sampleTransitioner/sampleContainer.jsx

```diff
@@ -34,203 +19,197 @@ export class SampleContainer extends React.Component {
   }
 
   isTransitionTarget(location) {
-    const { transitionTargets } = this.props;
+    const { transitionTargetLocations } = this.props;
 
     if (!this.isTargetContainer()) {
       return false;
     }
 
-    return !!transitionTargets.find(tl => tl.equals(location));
+    return !!transitionTargetLocations.find(tl => tl.equals(location));
   }
 
   isTransitionTargetOfHoveredSample(location) {
-    const { transitionTargetsOfHoveredSample } = this.props;
+    const { transitionTargetLocationsOfHoveredSample } = this.props;
 
     if (!this.isTargetContainer()) {
       return false;
     }
 
-    return !!transitionTargetsOfHoveredSample.find(tl => tl.equals(location));
+    return !!transitionTargetLocationsOfHoveredSample.find(tl => tl.equals(location));
   }
 
-  createRows() {
-    let rows = [];
-    let colsHeader = [];
-    let key;
-
-    key = '-1_-1';
-    colsHeader.push(<td key={key} style={cellStyleHeader} />);
-    for (let c = 0; c < this.props.cols; c++) {
-      key = `${c}_-1`;
-      colsHeader.push(
-        <td key={key} style={this.getHeaderStyle(-1, c)}>
-          {this.getColIndicator(c)}
-        </td>
-      );
+  onMouseOut() {
+    const { onMouseOut } = this.props;
+
+    if (this.state.hoverRow || this.state.hoverCol) {
+      this.setState({hoverRow: null, hoverCol: null});
     }
-    rows.push(<tr>{colsHeader}</tr>);
-
-    for (let r = 0; r < this.props.rows; r++) {
-      let cols = [];
-      // Get all samples with this row
-      const rowSamples = this.props.samples.filter(s => s.location.row === r);
-
-      key = `-1_${r}`;
-      cols.push(
-        <td key={key} style={this.getHeaderStyle(r, -1)}>
-          {this.getRowIndicator(r)}
-        </td>
+
+    onMouseOut();
+  }
+
+  renderColumnsHeader() {
+    const { numColumns } = this.props;
+    const keyPrefix = 'thead-th';
+    const ths = [(<th key={`${keyPrefix}-corner`} />)];
+
+    for (let c = 0; c < numColumns; c++) {
+      const label = c + 1;
+      ths.push(
+        <th key={`${keyPrefix}-${c}`} className={this.getHeaderClassName(-1, c)}>{label}</th>
       );
-      for (let c = 0; c < this.props.cols; c++) {
-        const thisLocation = new SampleLocation(this.props.id, c, r);
+    }
 
-        key = `${c}_${r}`;
-        const sample = rowSamples.find(s => s.location.col === c);
-        const sampleId = sample ? sample.id : null;
+    return (<tr key='thead-tr'>{ths}</tr>);
+  }
+
+  renderRowHeader(row) {
+    const keyPrefix = 'th';
+    const label = String.fromCharCode(65 + row);
+    return(<th key={`${keyPrefix}-${row}`} className={this.getHeaderClassName(row, -1)}>{label}</th>);
+  }
+
+  renderSampleWell(row, col, sampleId) {
+    const {
+      containerId,
+      onWellClicked,
+      onWellMouseOver,
+    } = this.props;
+
+    const key = `samplewell-${row}-${col}`;
+    const location = new SampleLocation(containerId, row, col);
+    const isTransitionSource = this.isTransitionSource(location);
```

Comment:
I wonder if some of these variables are unnecessarily set here, as they can be set in the return statement? Just thinking of readability since this indicates to my brain that this will be used in some way before returning.

Reply:
Fixed this now.